modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -360,19 +369,15 @@ public InputStream getInputStream(OperationPurpose purpose, String blob, long po
             totalSize = position + length;
         }
         BlobAsyncClient blobAsyncClient = asyncClient.getBlobContainerAsyncClient(container).getBlobAsyncClient(blob);
-        int maxReadRetries = service.getMaxReadRetries(projectId, clientName);
-        try {
-            return new AzureInputStream(
-                blobAsyncClient,
-                position,
-                length == null ? totalSize : length,
-                totalSize,
-                maxReadRetries,
-                azureBlobServiceClient.getAllocator()
-            );
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
+        return new AzureInputStream(
+            blobAsyncClient,
+            position,
+            length == null ? totalSize : length,
+            totalSize,
+            0,
```

Comment:
We disable Azure client retries in these downloads so we can control them in the `RetryingInputStream`